### ISSUE TITLE
DS Client and composite key

### ISSUE THREAD

**Issue by jfgosselin** [Impetus/kundera#552]
First thank you for adding the Datastax Java Driver client . Any plan to add support for Cassandra 2.x ?

I've started to use Kundera 2.11 and ran into some issues when a composite key is used. In DSClient when you iterate over the rows you should create a new 
compositeKeyInstance each time :

Instead of :

```
private List iterateAndReturn(ResultSet rSet, Class entityClazz, EntityMetadata metadata)
{
    // TODO:: Handle secondary table support case

    MetamodelImpl metaModel = (MetamodelImpl) kunderaMetadata.getApplicationMetadata().getMetamodel(
            metadata.getPersistenceUnit());
    EntityType entityType = metaModel.entity(metadata.getEntityClazz());

    Iterator<Row> rowIter = rSet.iterator();
    List results = new ArrayList();

    Map<String, Object> relationalValues = new HashMap<String, Object>();
    Map<String, Field> compositeColumns = new HashMap<String, Field>();

    Object compositeKeyInstance = null;
    if (metaModel.isEmbeddable(metadata.getIdAttribute().getBindableJavaType()))
    {
        compositeKeyInstance = getCompositeKeyInstance(metadata);
        EmbeddableType compositeKey = metaModel.embeddable(metadata.getIdAttribute().getBindableJavaType());
        Iterator<Attribute> attributes = compositeKey.getAttributes().iterator();
        while (attributes.hasNext())
        {
            Attribute attribute = attributes.next();
            String columnName = ((AbstractAttribute) attribute).getJPAColumnName();
            compositeColumns.put(columnName, (Field) attribute.getJavaMember());
        }
    }

    while (rowIter.hasNext())
    {
        Object entity = null;
        Row row = rowIter.next();
        ColumnDefinitions columnDefs = row.getColumnDefinitions();
        Iterator<Definition> columnDefIter = columnDefs.iterator();
```

You should have (I think) :

```
private List iterateAndReturn(ResultSet rSet, Class entityClazz, EntityMetadata metadata)
{
    // TODO:: Handle secondary table support case

    MetamodelImpl metaModel = (MetamodelImpl) kunderaMetadata.getApplicationMetadata().getMetamodel(
            metadata.getPersistenceUnit());
    EntityType entityType = metaModel.entity(metadata.getEntityClazz());

    Iterator<Row> rowIter = rSet.iterator();
    List results = new ArrayList();

    Map<String, Object> relationalValues = new HashMap<String, Object>();
    Map<String, Field> compositeColumns = new HashMap<String, Field>();

    while (rowIter.hasNext())
    {
    Object compositeKeyInstance = null;
    if (metaModel.isEmbeddable(metadata.getIdAttribute().getBindableJavaType()))
    {
        compositeKeyInstance = getCompositeKeyInstance(metadata);
        EmbeddableType compositeKey = metaModel.embeddable(metadata.getIdAttribute().getBindableJavaType());
        Iterator<Attribute> attributes = compositeKey.getAttributes().iterator();
        while (attributes.hasNext())
        {
            Attribute attribute = attributes.next();
            String columnName = ((AbstractAttribute) attribute).getJPAColumnName();
            compositeColumns.put(columnName, (Field) attribute.getJavaMember());
        }
    }

        Object entity = null;
        Row row = rowIter.next();
        ColumnDefinitions columnDefs = row.getColumnDefinitions();
        Iterator<Definition> columnDefIter = columnDefs.iterator();
```

**Comment by mevivs** [comment]
Fix has been added for this issue.

Thanks,
-Vivek

**Comment by chhavigangwal** [comment]
Fixed in Kundera-2.11.1
